.github/workflows/doxygen.yaml

```
@@ -16,8 +16,113 @@ name: Doxygen
 
 on:
     push:
+        paths:
+            - '**.dox'
+            - '**.doc'
+            - '**.c'
+            - '**.cc'
+            - '**.cxx'
+            - '**.cpp'
+            - '**.c++'
+            - '**.ii'
+            - '**.ixx'
+            - '**.ipp'
+            - '**.i++'
+            - '**.inl'
+            - '**.h'
+            - '**.H'
+            - '**.hh'
+            - '**.HH'
+            - '**.hxx'
+            - '**.hpp'
+            - '**.h++'
+            - '**.mm'
+            - '**.txt'
+            - '**.idl'
+            - '**.ddl'
+            - '**.odl'
+            - '**.java'
+            - '**.cs'
+            - '**.d'
+            - '**.php'
+            - '**.php4'
+            - '**.php5'
+            - '**.inc'
+            - '**.phtml'
+            - '**.m'
+            - '**.M'
```

@andy31415 this is my initial attempt the use the [GH match filters](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#patterns-to-match-file-paths), unfortunately I don't see examples of grouping characters in the regex. 

```suggestion
            - '**.do[xc]'
            - '**.c[cs]?'
            - '**.cxx'
            - '**.cpp'
            - '**.c++'
            - '**.ii'
            - '**.ixx'
            - '**.ipp'
            - '**.i++'
            - '**.inl'
            - '**.[hH]'
            - '**.hh'
            - '**.HH'
            - '**.hxx'
            - '**.hpp'
            - '**.h++'
            - '**.mm'
            - '**.txt'
            - '**.[ido]dl'
            - '**.java'
            - '**.d'
            - '**.php[45]?'
            - '**.inc'
            - '**.phtml'
            - '**.[mM]'
```